### ISSUE TITLE
Training loss extremely noisy during fine-tuning and randomly goes to 0

### ISSUE THREAD

**Issue by zpx01** [deepseek-ai/DeepSeek-Coder#106]
I'm trying to fine-tune the 6.7B model on my own code dataset. I am running a multinode training with fp32 precision on NVIDIA Tesla V100 GPUs with DeepSpeed ZeRO Stage 3. My training loss seems to randomly fluctuate and go down to zero, I've attached my training loss graph below:

<img width="513" alt="Screenshot 2024-01-25 at 10 19 48 PM" src="https://github.com/deepseek-ai/DeepSeek-Coder/assets/51731412/ccbda14d-4c87-4b50-853d-e46f8ccb195a">

I'm running this on 128 GPUs with a train batch size of 1 per device and no gradient accumulation. I'm not sure what could be the cause of this as I haven't seen this happen with other models with the Llama architecture. Would appreciate any general direction to help debug this, thanks!

**Comment by zpx01** [comment]
@DejianYang @pkuzqh Would appreciate any help on this ticket, thanks

**Comment by Locher7** [comment]
How did you solve it?
